packages/svelte/tests/runtime-runes/samples/non-local-mutation-inherited-owner/main.svelte

```diff
@@ -2,8 +2,7 @@
 	import Counter from './Counter.svelte';
 	import { global } from './state.svelte.js';
 
-	let object = $state({ count: 0 });
-	global.object = object;
+	global.object = { count: 0 };
```

Comment:
addressed this above:

> ...the parent of `object` is still `null`, _not_ `some_other_state`. This means one of the tests needed adjusting, but I think that's okay.

We _could_ make it work with a bit of finagling but I'm not convinced we should. I don't think you'd ever need to write code like that in real life, I think the mental model is clearer in this PR

Reply:
I don't know, feels like it's only a matter of time until some combination of reassignments will lead to such a situation. Could we do something like "when state is assigned to a different state, lift ownership of assigned state completely"? You're saying this situation is super rare, so this would trade a rare false positive (bad) with a rare false negative (okay).

Reply:
But what if you have a case like this instead?

```diff
-let object = $state({ count: 0 });
-global.object = object;
+let object = $state({ nested: { count: 0 } });
+global.nested = object.nested;
```

Does the parent of `object.nested` change from `object` to `global`? Or at that point do we need to allow things to have multiple parents, with all the extra complexity (and memory leak risk) that implies? Just seems like the dictionary definition of YAGNI to me.

Reply:
The parent doesn't change at all, it stays the same. But whatever defines the allowed set of owners switches from "this set of components" to "everything" for `object.nested`.

Reply:
I see. It would definitely add some complexity (we would need a new special value for indicating universal ownership rather than using `null` to indicate 'not owned by anyone', and we'd need a `get_all_owners` function), so my inclination is still to file it under YAGNI. If it happens in the wild, we could revisit but it doesn't feel like something we should be defensive about

Reply:
I pushed a commit implementing what I had in mind - it's only a handful lines of code